qa/tasks/cephfs/test_misc.py

```diff
@@ -232,35 +233,63 @@ def test_fs_lsflags(self):
         self.assertEqual(lsflags["allow_multimds_snaps"], True)
         self.assertEqual(lsflags["allow_standby_replay"], True)
 
-    def test_filesystem_sync_stuck_for_around_5s(self):
-        """
-        To check whether will the fsync will be be stuck to wait for the mdlog to be
-        flushed for at most 5 seconds.
-        """
+    def _test_sync_stuck_for_around_5s(self, dir_path, file_sync=False):
 
-        dir_path = "fsync_do_not_wait_mdlog_testdir"
         self.mount_a.run_shell(["mkdir", dir_path])
 
+        pyscript = dedent("""
+                import os
+
+                path = "{path}"
+                dfd = os.open(path, os.O_DIRECTORY)
+                os.fsync(dfd)
+                os.close(dfd)
+            """.format(path=dir_path))
+
         # run create/delete direcoties and test the sync time duration
         for i in range(300):
             for j in range(5):
                 self.mount_a.run_shell(["mkdir", os.path.join(dir_path, f"{i}_{j}")])
             start = time.time()
-            self.mount_a.run_shell(["sync"])
+            if file_sync:
+                self.mount_a.run_shell(['python3', '-c', pyscript])
+            else:
+                self.mount_a.run_shell(["sync"])
             duration = time.time() - start
-            log.info(f"mkdir i = {i}, duration = {duration}")
+            log.info(f"sync mkdir i = {i}, duration = {duration}")
             self.assertLess(duration, 4)
 
             for j in range(5):
                 self.mount_a.run_shell(["rm", "-rf", os.path.join(dir_path, f"{i}_{j}")])
             start = time.time()
-            self.mount_a.run_shell(["sync"])
+            if file_sync:
+                self.mount_a.run_shell(['python3', '-c', pyscript])
+            else:
+                self.mount_a.run_shell(["sync"])
             duration = time.time() - start
-            log.info(f"rmdir i = {i}, duration = {duration}")
+            log.info(f"sync rmdir i = {i}, duration = {duration}")
             self.assertLess(duration, 4)
 
         self.mount_a.run_shell(["rm", "-rf", dir_path])
 
+    def test_filesystem_sync_stuck_for_around_5s(self):
+        """
+        To check whether will the fsync will be be stuck to wait for the mdlog to be
+        flushed for at most 5 seconds.
+        """
+
+        dir_path = "filesystem_sync_do_not_wait_mdlog_testdir"
+        self._test_sync_stuck_for_around_5s(dir_path)
+
+    def test_file_sync_stuck_for_around_5s(self):
+        """
+        To check whether will the filesystem sync will be be stuck to wait for the
```

Comment:
nit: s/"To check whether will the fsync will be be stuck to wait"/"To check whether the fsync will be stuck waiting"/